app/scripts/controllers/network-set-controller.js

```diff
@@ -56,6 +64,49 @@ ndexApp.controller('networkSetController',
         $scope.copyNetworkSetURLTitle = 'Copied';
     };
 
+
```

Comment:
This function has some repeated code. Could you rewrite it to get rid of the repeated code? Could you pull the error handler out and apply it in all the API calls that are used in this page? Or each API call needs a customized error handler?

Reply:
If the user has an empty set, the new get summaries function still calls the server once. It should not call the server according to Vlad's design document.